clientserver/src/mill/clientserver/Client.java

```diff
@@ -100,16 +103,26 @@ public static int run(String lockBase,
         FileOutputStream f = new FileOutputStream(lockBase + "/run");
         ClientServer.writeArgs(System.console() != null, args, f);
         f.close();
-        if (locks.processLock.probe()) initServer.run();
+
+        boolean serverInit = false;
+        if (locks.processLock.probe()) {
+            serverInit = true;
+            initServer.run();
+        }
         while(locks.processLock.probe()) Thread.sleep(3);
 
+        // Need to give sometime for Win32NamedPipeSocket to work
+        // if the server is just initialized
+        if (serverInit && ClientServer.isWindows) Thread.sleep(250);
 
-        UnixDomainSocket ioSocket = null;
+        Socket ioSocket = null;
 
         long retryStart = System.currentTimeMillis();
         while(ioSocket == null && System.currentTimeMillis() - retryStart < 1000){
             try{
-                ioSocket = new UnixDomainSocket(lockBase + "/io");
+                ioSocket = ClientServer.isWindows?
+                        new Win32NamedPipeSocket("\\\\.\\pipe\\" + new File(lockBase).getName())
```

Comment:
Can we put the `\\\\.\\pipe\\` string into a static constant in `ClientServer` so we don't have to copy paste it? That will also give us a place to add a short comment with a link to the windows named-pipe doc where this magic-string comes from

Reply:
Sure.